compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SyntheticAccessorLowering.kt

```diff
@@ -130,34 +146,49 @@ private class SyntheticAccessorLowering(val context: JvmBackendContext) : IrElem
             copyAllParamsToArgs(it, accessor)
         }
 
-    private fun IrSimpleFunction.makeSimpleFunctionAccessor(): IrSimpleFunction {
+    private fun IrSimpleFunction.makeSimpleFunctionAccessor(superQualifierSymbol: IrClassSymbol?): IrSimpleFunction {
         val source = this
         return buildFun {
             origin = JvmLoweredDeclarationOrigin.SYNTHETIC_ACCESSOR
             name = source.accessorName()
             visibility = Visibilities.PUBLIC
             isSuspend = source.isSuspend
         }.also { accessor ->
-            accessor.parent = source.parent
+            accessor.parent = if ((source.parent as? IrClass)?.isInterface == true) {
+                // Before Java 8, interfaces can't contain implementations.
+                // Place accessor in DefaultImpls for interfaces.
+                // TODO: remove this after targeting Java 8 or newer.
```

Comment:
NB: Accessors in interfaces could be only for private interface methods, they should be placed always in DefaulImpls, except cases when private methods annotated with @JvmDefault (only in such cases JVM default method is generated)

Reply:
Updated the comments accordingly. Please take a look again.